src/main/java/com/facebook/openwifirrm/modules/DataCollector.java

```diff
@@ -260,6 +260,15 @@ private void fetchDeviceCapabilities(List<DeviceWithStatus> devices) {
 			Math.max(params.capabilitiesIntervalSec, 1) * 1_000_000_000L;
 
 		for (DeviceWithStatus device : devices) {
+			// Check if online/connected
```

Comment:
I think the code `!device.connected` is self-explanatory, so you don't need the comment. Up to you. Same comment for the other instances of `!device.connected`

Reply:
I like to keep comments above nearly every "paragraph" of code so it's quicker to skim files without actually reading any code.